### PR TITLE
More CMake fixes

### PR DESCRIPTION
**Pull request by MaartenBent**
Some more CMake fixes of issues I encountered when trying it on Linux.

The GTK build was missing the opengl files, causing missing symbols when building the opengl samples.

Building `test_gui` failed because `-lmedia` was missing (because `wxUSE_MEDIACTRL 0`). I added checks for all libraries used in `wx_exe_link_libraries` so they are only added when enabled.

### CODE REVIEW

**Review by TcT2k — CHANGES_REQUESTED** [review]
You should add the missing files to build/files and run upmake afterwards so the changes won't be overwritten

**Comment by MaartenBent** [comment]
Done, but running upmake didn't work as expected.

It only seems to work when the variables are already present in `files.cmake` (`OPENGL_GTK_SRC` and `OPENGL_GTK_HDR`). After adding them manually, upmake will add the files.

**Comment by vadz** [comment]
Yes, upmake never adds new variables, it only updates the contents of the existing ones. It's supposed to do one thing and one thing only: allow to have the list of source files (and maybe headers) in one place and update all the other make/project files automatically and is, by design, too dumb to do anything else.

**Comment by MaartenBent** [comment]
Rebased and added `uxtheme` library.

File: build/cmake/functions.cmake [hunk 1]
```diff
@@ -166,6 +166,9 @@ function(wx_set_target_properties target_name is_base)
     endif()
 
     if(wxUSE_UNICODE)
+        if(WIN32)
```
Comment (vadz) [inline build/cmake/functions.cmake]: Curious, is this really necessary? Defining just `_UNICODE` is supposed to be enough -- even if defining this one won't do any real harm neither.

**Review by vadz — APPROVED** [review]
Looks good to me, thanks!

@TcT2k I'll merge this soon if you don't object.

**Review by TcT2k — APPROVED** [review]
Looks good to me

File: build/cmake/functions.cmake (hunk 1 above)
Comment (MaartenBent) [reply build/cmake/functions.cmake]: Yes, the Windows SDK headers require `UNICODE`, the c++ headers require `_UNICODE`.
Normally wxWidgets will define both if one of the two is defined (in `platform.h`), and Visual Studio also adds them by default. But `sdk_exe` doesn't include any wxWidgets headers, and MinGW does not automatically define `_UNICODE`, hence this fix.

**Pull request merged**